detection_rules/rule_validators.py

```diff
@@ -713,28 +722,348 @@ def validate_rule_type_configurations(self, data: EQLRuleData, meta: RuleMeta) -
 class ESQLValidator(QueryValidator):
     """Validate specific fields for ESQL query event types."""
 
+    def __init__(self, query: str) -> None:
+        """Initialize the ESQLValidator with the given query."""
+        super().__init__(query)
+        self.esql_unique_fields: list[dict[str, str]] = []
+        self.related_integrations: list[dict[str, str]] = []
+        self.stack_version: str = ""
+
     @cached_property
     def ast(self) -> None:  # type: ignore[reportIncompatibleMethodOverride]
+        """There is no AST for ESQL until we have an ESQL parser."""
         return None
 
     @cached_property
     def unique_fields(self) -> list[str]:  # type: ignore[reportIncompatibleMethodOverride]
-        """Return a list of unique fields in the query."""
-        # return empty list for ES|QL rules until ast is available (friendlier than raising error)
+        """Return a list of unique fields in the query. Requires remote validation to have occurred."""
+        if self.esql_unique_fields:
+            return [field["name"] for field in self.esql_unique_fields]
         return []
 
-    def validate(self, _: "QueryRuleData", __: RuleMeta) -> None:  # type: ignore[reportIncompatibleMethodOverride]
+    def get_unique_field_type(self, field_name: str) -> str | None:  # type: ignore[reportIncompatibleMethodOverride]
+        """Get the type of the unique field. Requires remote validation to have occurred."""
+        for field in self.esql_unique_fields:
+            if field["name"] == field_name:
+                return field["type"]
+        return None
+
+    def validate(self, rule_data: "QueryRuleData", rule_meta: RuleMeta, force_remote_validation: bool = False) -> None:  # type: ignore[reportIncompatibleMethodOverride]
         """Validate an ESQL query while checking TOMLRule."""
-        # temporarily override to NOP until ES|QL query parsing is supported
+        if misc.getdefault("remote_esql_validation")() or force_remote_validation:
+            kibana_client = misc.get_default_kibana_client()
+            elastic_client = misc.get_default_elasticsearch_client()
+            _ = self.remote_validate_rule(
+                kibana_client,
+                elastic_client,
+                rule_data.query,
+                rule_meta,
+                rule_data.rule_id,
+            )
 
-    def validate_integration(
+    def get_rule_integrations(self, metadata: RuleMeta) -> list[str]:
```

Comment:
I like the introduction of this. It would be nice to double check where we could use this throughout the repo.